lib/puma/launcher.rb

```diff
@@ -48,18 +49,28 @@ def initialize(conf, launcher_args={})
 
       env = launcher_args.delete(:env) || ENV
 
-      @config.options[:log_writer] = @log_writer
+      @config.load
+      @config.clamp
+      @options = @config.options
+
+      @options[:log_writer] = @log_writer
+      @options[:logger] = @log_writer if clustered?
+      if @options[:bind_to_activated_sockets]
+        @options[:binds] = @binder.synthesize_binds_from_activated_fs(
+          @options[:binds],
+          @options[:bind_to_activated_sockets] == 'only'
+        )
+      end
 
       # Advertise the Configuration
       Puma.cli_config = @config if defined?(Puma.cli_config)
+      log_config if env['PUMA_LOG_CONFIG']
```

Comment:
I've QA'ed to ensure moving this here doesn't actually change the position of the configuration in the logs. It's just nicer to group all the config related stuff together.